azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -352,35 +410,37 @@ def _parse_plugin_settings(ext_handler, plugin_settings):
         runtime_settings_node = find(settings[0], "RuntimeSettings")
         seqNo = getattrib(runtime_settings_node, "seqNo")
         runtime_settings_str = gettext(runtime_settings_node)
-        try:
-            runtime_settings = json.loads(runtime_settings_str)
-        except ValueError as e:
-            logger.error("Invalid extension settings")
-            return
+        if runtime_settings_str is not None:
```

Comment:
the previous code assumed this property was always present and would have raised an exception if it wasn't. What are the implications of adding this check?

Reply:
See answer below